### DISCUSSION TITLE
Does PrometheusHttpServer support gzip compression, and how can I configure it

### DISCUSSION THREAD

**Discussion by GenerousMan** [open-telemetry/opentelemetry-java#6114] [Q&A] [answered]
I'm exposing metrics on port 9090 with PrometheusHttpServer, but the sheer volume of data is really straining my network bandwidth. I'm thinking about using gzip compression to ease up on the network load. Right now, I'm on the 1.29.0-alpha version, and I noticed there's a shouldUseCompression method in there. Is there a way to configure this compression setting?

**Comment by jack-berg** [comment, marked as answer]
PrometheusHttpServer will gzip compress the content if the request includes `Accept-Encoding: gzip`

**Reply by GenerousMan** [reply]
Thank you very much for your response. When I instantiate a PrometheusHttpServer, I am able to configure its host and port using the builder pattern as follows: `PrometheusHttpServer.builder().setHost().setPort()`. The metricsHandler within this setup retrieves the headers from the HttpExchange using `exchange.getRequestHeaders().get("Accept-Encoding")` to determine whether gzip compression should be enabled.

However, upon reviewing the latest iteration of the codebase, it appears that this particular segment of code has been omitted. Could you please confirm if this functionality remains operable, and if so, advise on the method for setting the requestHeaders for the `HttpExchange`?

**Reply by jack-berg** [reply]
Yup it still works. #6015 refactored the prometheus exporter to depend on the upstream prometheus client library for serialization and serving requests. The goal was to reduce maintenance burden, add support for exponential histograms, and make it more easy to add various additional features. Gzip compression is still supported.
